app/helpers/samples_helper.rb

```diff
@@ -122,8 +122,30 @@ def samples_pipeline_run_info(samples)
     pipeline_run_info
   end
 
-  def samples_info(samples)
-    { final_result: samples_output_data(samples),
-      pipeline_run_info: samples_pipeline_run_info(samples) }
+  def filter_samples(samples, query)
+    filtered = []
```

Comment:
You can try something like
if query == 'UPLOADING' 
  samples = samples.where(pipeline_run_id: nil)
else
samples = samples..where(“pipeline_id in (select max(pipeline_id) from pipeline_runs group by sample_id”).where(job_status: filter_qury) 
end